minicluster/src/main/java/tachyon/master/LocalTachyonCluster.java

```diff
@@ -115,88 +134,117 @@ public NetAddress getWorkerAddress() {
-    mWorkerConf.set(String.format(Constants.WORKER_TIERED_STORE_LEVEL_ALIAS_FORMAT, 0), "MEM");
-    mWorkerConf.set(String.format(Constants.WORKER_TIERED_STORE_LEVEL_DIRS_PATH_FORMAT, 0),
+    mTestConf.set(String.format(Constants.WORKER_TIERED_STORE_LEVEL_ALIAS_FORMAT, 0), "MEM");
+    mTestConf.set(String.format(Constants.WORKER_TIERED_STORE_LEVEL_DIRS_PATH_FORMAT, 0),
         ramdiskPath);
-    mWorkerConf.set(String.format(Constants.WORKER_TIERED_STORE_LEVEL_DIRS_QUOTA_FORMAT, 0),
+    mTestConf.set(String.format(Constants.WORKER_TIERED_STORE_LEVEL_DIRS_QUOTA_FORMAT, 0),
         Long.toString(mWorkerCapacityBytes));
-    UnderFileSystemUtils.mkdirIfNotExists(ramdiskPath, mWorkerConf);
+    UnderFileSystemUtils.mkdirIfNotExists(ramdiskPath, mTestConf);
 
-    int maxLevel = mWorkerConf.getInt(Constants.WORKER_TIERED_STORAGE_LEVEL_MAX);
+    int maxLevel = mTestConf.getInt(Constants.WORKER_TIERED_STORAGE_LEVEL_MAX);
     for (int level = 1; level < maxLevel; level ++) {
       String tierLevelDirPath =
           String.format(Constants.WORKER_TIERED_STORE_LEVEL_DIRS_PATH_FORMAT, level);
-      String[] dirPaths = mWorkerConf.get(tierLevelDirPath).split(",");
+      String[] dirPaths = mTestConf.get(tierLevelDirPath).split(",");
       List<String> newPaths = new ArrayList<String>();
       for (String dirPath : dirPaths) {
         String newPath = mTachyonHome + dirPath;
         newPaths.add(newPath);
-        UnderFileSystemUtils.mkdirIfNotExists(newPath, mWorkerConf);
+        UnderFileSystemUtils.mkdirIfNotExists(newPath, mTestConf);
       }
-      mWorkerConf.set(String.format(Constants.WORKER_TIERED_STORE_LEVEL_DIRS_PATH_FORMAT, level),
+      mTestConf.set(String.format(Constants.WORKER_TIERED_STORE_LEVEL_DIRS_PATH_FORMAT, level),
           Joiner.on(',').join(newPaths));
     }
+  }
 
-    // We need to update the client with the most recent configuration so it knows the correct ports
+  public TachyonConf getTestConf() {
+    return mTestConf;
+  }
+
+  /**
+   * Configures and starts master.
+   *
+   * @throws IOException when the operation fails
+   */
+  private void startMaster() throws IOException {
+    mMasterConf = new TachyonConf(mTestConf.getInternalProperties());
+    MasterContext.reset(mMasterConf);
+
+    mMaster = LocalTachyonMaster.create(mTachyonHome);
+    mMaster.start();
+
+    // Update the test conf with actual RPC port.
+    mTestConf.set(Constants.MASTER_PORT, String.valueOf(getMasterPort()));
+  }
+
+  /**
+   * Configure and start worker.
+   *
+   * @throws IOException when the operation fails
+   */
+  private void startWorker() throws IOException {
+    mWorkerConf = new TachyonConf(mTestConf.getInternalProperties());
+    mClientConf = new TachyonConf(mTestConf.getInternalProperties());
+    WorkerContext.reset(mWorkerConf);
+    ClientContext.reset(mClientConf);
 
+    // We need to update the client with the most recent configuration so it knows the correct ports
```

Comment:
`correct ports` of what? If the `ClientContext` only needs to update its knowledge of `MASTER_PORT`, why not put related logic into `startMaster()`?